components/miro_custom_app/actions/update-shape/update-shape.mjs

```diff
@@ -0,0 +1,48 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  name: "Update Shape",
+  version: "0.0.2",
+  key: "update-shape",
+  description: "Updates content of an existing shape on a Miro board",
+  type: "action",
+  props: {
+    miro_custom_app: {
+      type: "app",
+      app: "miro_custom_app"
+    },
+    boardId: {
+      type: "string",
+      description: "Board ID",
+      optional: false,
+    },
+    itemId: {
+      type: "string",
+      description: "Shape (item) ID",
+      optional: false,
+    },
```

Comment:
Here you can create an `async options()` prop to get a dropdown menu to select both a `boardId` and an `itemId` using the api.